packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -428,6 +428,137 @@ abstract class TransitionRoute<T> extends OverlayRoute<T> {
+  /// Similarly when the topmost route is popped, the secondaryAnimation
+  /// can be used to define how the previous route below it reappears on the screen.
+  ///
+  /// When the Navigator pushes this new route on the top of its stack,
+  /// the old topmost route's secondaryAnimation runs from 0.0 to 1.0.
+  /// When the Navigator pops the topmost route, the
+  /// secondaryAnimation for the route below it runs from 1.0 to 0.0.
+  ///
+  /// The example below adds a transition that's driven by the
+  /// [secondaryAnimation]. When the previous route disappears because this route has
+  /// been pushed on top of it, it translates to right. And the opposite when
+  /// the route is exposed because this topmost route has been popped off.
+  ///
+  /// ```dart
+  /// return SlideTransition(
+  ///   position: TweenOffset(
+  ///     begin: Offset.zero,
+  ///     end: const Offset(0.0, 1.0),
+  ///   ).animate(secondaryAnimation),
+  ///   child: child,
+  /// );
+  /// ```
+  ///
+  /// In practice this method is used pretty rarely.
+  ///
+  /// The arguments to this method are as follows:
+  ///
+  ///  * `context`: The context in which the route is being built.
+  ///  * [secondaryAnimation]: When the Navigator pushes this route
+  ///    on the top of its stack, the previous topmost route's [secondaryAnimation]
+  ///    runs from 0.0 to 1.0. When the [Navigator] pops the topmost route, the
+  ///    [secondaryAnimation] for the route below it runs from 1.0 to 0.0.
+  ///  * `child`, the page contents from the previous route by
+  ///     previous route's [buildPage].
+  Widget buildSecondaryAnimationTransitionForPreviousRoute(
+    BuildContext context,
+    Animation<double> secondaryAnimation,
+    Widget child,
+  ) {
+    return child;
+  }
+
+  /// Returns true if the [previousRoute] bottom route should delegate the
+  /// `secondaryAnimation` transition when this route is pushed on top of it
+  /// or when then this route is popped off of it.
+  ///
+  /// Subclasses can override this method to restrict the set of routes they
+  /// need to coordinate transitions with.
+  ///
+  /// If true, and `previousRoute.canTransitionTo()` is true, then the route
+  /// [previousRoute] will delegate its `secondaryAnimation` transition to
+  /// this route's [buildSecondaryAnimationTransitionForPreviousRoute] method
+  ///
+  /// If false, [previousRoute]'s `secondaryAnimation` transition will be
+  /// handled by default by [buildTransitions]
+  ///
+  /// Returns false by default.
+  ///
+  /// See also:
+  ///  * [canTransitionTo], which must be true for [previousRoute] for the
+  ///    [getSecondaryAnimationTransitionForPreviousRoute] `secondaryAnimation` to run.
+  ///
+  ///  * [buildSecondaryAnimationTransitionForPreviousRoute], to define
+  ///    [previousRoute]'s `secondaryAnimation` transition when this is true
+  bool handleSecondaryAnimationTransitionForPreviousRoute(Route<dynamic> previousRoute) => false;
```

Comment:
Maybe: canDriveSecondaryTransitionForPreviousRoute (analogous to canTransitionTo)